ydb/core/tx/schemeshard/schemeshard__operation_common_resource_pool.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "schemeshard__operation_part.h"
+#include "schemeshard_path.h"
+
+#define LOG_I(stream) LOG_INFO_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD, "[" << context.SS->TabletID() << "] " << stream)
+#define LOG_N(stream) LOG_NOTICE_S(context.Ctx, NKikimrServices::FLAT_TX_SCHEMESHARD, "[" << context.SS->TabletID() << "] " << stream)
+#define RETURN_RESULT_UNLESS(x) if (!(x)) return result;
+
+
+namespace NKikimr::NSchemeShard {
+
+namespace NResourcePool {
+
+TPath::TChecker IsParentPathValid(const TPath& parentPath);
+
+bool IsParentPathValid(const THolder<TProposeResponse>& result, const TPath& parentPath);
+
+bool Validate(const NKikimrSchemeOp::TResourcePoolDescription& description, TString& errorStr);
+
+TResourcePoolInfo::TPtr CreateResourcePool(const NKikimrSchemeOp::TResourcePoolDescription& description, ui64 alterVersion);
+
+TResourcePoolInfo::TPtr ModifyResourcePool(const NKikimrSchemeOp::TResourcePoolDescription& description, const TResourcePoolInfo::TPtr oldResourcePoolInfo);
+
+}  // namespace NResourcePool
+
+class TResourcePoolSubOperation : public TSubOperation {
+public:
+    using TSubOperation::TSubOperation;
+
+protected:
+    static TTxState::ETxState NextState();
+    virtual TTxState::ETxState NextState(TTxState::ETxState state) const override;
+    virtual TSubOperationState::TPtr SelectStateFunc(TTxState::ETxState state) override;
+    virtual TSubOperationState::TPtr GetProposeOperationState() = 0;
```

Comment:
This will move state graph definition far away from final `TSubOperation` object, hiding necessary and valuable information.